packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -138,33 +143,35 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 	): void {
 		const postUpdateActions: (() => void)[] = [];
 		const audienceMembers = this.audience.getMembers();
+		const joiningAttendees = new Set<SessionClient>();
 		const connectedAttendees = new Set<SessionClient>();
 		for (const [clientConnectionId, value] of Object.entries(
 			remoteDatastore.clientToSessionId,
 		)) {
 			const clientSessionId = value.value;
-			const { attendee, isNew } = this.ensureAttendee(
+			const { attendee, isJoining } = this.ensureAttendee(
 				clientSessionId,
 				clientConnectionId,
 				/* order */ value.rev,
 			);
 
-			// Check new attendee against audience to see if they're currently connected
-			const isAttendeeConnected = audienceMembers.has(clientConnectionId);
+			const isAttendeeConnected =
+				// Attendee is connected if they are present in audience
+				audienceMembers.has(clientConnectionId) ||
+				// Attendee is connected if they are the sender of the update signal
+				senderConnectionId === clientConnectionId ||
+				// Attendee is connected if they were already marked as connected
+				connectedAttendees.has(attendee);
 
 			if (isAttendeeConnected) {
+				// If attendee is connected, update their connection ID and status.
 				connectedAttendees.add(attendee);
-				if (attendee.getConnectionStatus() === SessionClientStatus.Disconnected) {
-					attendee.setConnectionId(clientConnectionId);
-				}
-				if (isNew) {
-					// If the attendee is both new and in audience (i.e. currently connected), emit an attendeeJoined event.
-					postUpdateActions.push(() => this.events.emit("attendeeJoined", attendee));
+				attendee.setConnectionId(clientConnectionId);
```

Comment:
Doesn't `ensureAttendee` already do connection ID update?
Why are any of the clientConnectionIds that happen to be associated with the attendee okay to set as the current connection id.
I guess we need a test case where early connection ID is in audience, but a later one processed is not in the audience. Result should be that the early actual connected ID is the one that is should be provided by getConnectionId().

Reply:
`ensureAttendee`makes sure that the correct connection ID is set (updateStatus is set to false), but this function is responsible for making sure the attendees connection status is set to 'Connected'. I updated the function to just setConnected() rather than messing around with assigning correct connectionID's.